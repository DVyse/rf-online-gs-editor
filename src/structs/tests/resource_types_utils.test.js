import {
  BONE_TYPES,
  MESH_TYPES,
  ANI_TYPES,
  isBone,
  isMesh,
  isAni,
} from '../resource_types_utils';

describe('Structs', () => {
  describe('ResourceTypesUtils', () => {
    it('should have boneTypes', () => {
      expect(BONE_TYPES).toMatchSnapshot();
    });
    it('should have meshTypes', () => {
      expect(MESH_TYPES).toMatchSnapshot();
    });
    it('should have aniTypes', () => {
      expect(ANI_TYPES).toMatchSnapshot();
    });
    it('should check isBone/isMesh/isAni', () => {
      [...BONE_TYPES, MESH_TYPES, ANI_TYPES].forEach(type => {
        expect(isBone(type)).toBe(BONE_TYPES.indexOf(type) !== -1);
        expect(isMesh(type)).toBe(MESH_TYPES.indexOf(type) !== -1);
        expect(isAni(type)).toBe(ANI_TYPES.indexOf(type) !== -1);
      });
    });
  });
});
