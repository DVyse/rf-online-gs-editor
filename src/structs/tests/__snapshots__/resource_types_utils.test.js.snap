// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Structs ResourceTypesUtils should have aniTypes 1`] = `
Array [
  "player_ani1",
  "player_ani2",
  "player_ani3",
  "monster_ani",
  "animus_ani",
  "tower_ani",
  "store_ani",
  "item_ani",
  "mau_ani",
]
`;

exports[`Structs ResourceTypesUtils should have boneTypes 1`] = `
Array [
  "player_bone1",
  "player_bone2",
  "player_bone3",
  "monster_bone",
  "animus_bone",
  "tower_bone",
  "store_bone",
  "item_bone",
  "mau_bone",
]
`;

exports[`Structs ResourceTypesUtils should have meshTypes 1`] = `
Array [
  "player_mesh1",
  "player_mesh2",
  "player_mesh3",
  "monster_mesh",
  "animus_mesh",
  "tower_mesh",
  "store_mesh",
  "item_mesh",
  "mau_mesh",
]
`;
